libcperciva/datastruct/elasticarray.c

```diff
@@ -30,7 +30,7 @@ resize(struct elasticarray * EA, size_t nsize)
 		nalloc = EA->alloc * 2;
 		if (nalloc < nsize)
 			nalloc = nsize;
-	} else if (EA->alloc > nsize * 4) {
+	} else if (EA->alloc / 4 > nsize) {
```

Comment:
Agreed on the code!  Just a few nitpicks in the commit message.  Again, the original one:

> In the case of modulo wrapping of nsize * 4, the else if block could be
> taken even though EA->alloc is not really greater than nsize * 4. This
> can lead to nalloc becoming smaller than nsize which will result in the
> buffer to be reallocated to be smaller than the desired nsize.

- two spaces after a period, please
- missing comma: "smaller than nsize, which will"